internal/config/config.go

```diff
@@ -461,40 +388,44 @@ func parseAddressPool(p addressPool, bgpCommunities map[string]uint32) (*Pool, e
 	return ret, nil
 }
 
-func parseBFDProfile(p bfdProfile) (*BFDProfile, error) {
+func bfdProfileFromCR(p metallbv1beta1.BFDProfile) (*BFDProfile, error) {
 	if p.Name == "" {
 		return nil, fmt.Errorf("missing bfd profile name")
 	}
 	res := &BFDProfile{}
 	res.Name = p.Name
 	var err error
-	res.DetectMultiplier, err = bfdIntFromConfig(p.DetectMultiplier, 2, 255)
+	res.DetectMultiplier, err = bfdIntFromConfig(p.Spec.DetectMultiplier, 2, 255)
 	if err != nil {
 		return nil, errors.Wrap(err, "invalid detect multiplier value")
 	}
-	res.ReceiveInterval, err = bfdIntFromConfig(p.ReceiveInterval, 10, 60000)
+	res.ReceiveInterval, err = bfdIntFromConfig(p.Spec.ReceiveInterval, 10, 60000)
 	if err != nil {
 		return nil, errors.Wrap(err, "invalid receive interval value")
 	}
-	res.TransmitInterval, err = bfdIntFromConfig(p.TransmitInterval, 10, 60000)
+	res.TransmitInterval, err = bfdIntFromConfig(p.Spec.TransmitInterval, 10, 60000)
 	if err != nil {
 		return nil, errors.Wrap(err, "invalid transmit interval value")
 	}
-	res.MinimumTTL, err = bfdIntFromConfig(p.MinimumTTL, 1, 254)
+	res.MinimumTTL, err = bfdIntFromConfig(p.Spec.MinimumTTL, 1, 254)
 	if err != nil {
 		return nil, errors.Wrap(err, "invalid minimum ttl value")
 	}
-	res.EchoInterval, err = bfdIntFromConfig(p.EchoInterval, 10, 60000)
+	res.EchoInterval, err = bfdIntFromConfig(p.Spec.EchoInterval, 10, 60000)
 	if err != nil {
 		return nil, errors.Wrap(err, "invalid minimum ttl value")
 	}
-	res.EchoMode = p.EchoMode
-	res.PassiveMode = p.PassiveMode
+	if p.Spec.EchoMode != nil {
+		res.EchoMode = *p.Spec.EchoMode
+	}
+	if p.Spec.PassiveMode != nil {
+		res.PassiveMode = *p.Spec.PassiveMode
+	}
 
 	return res, nil
 }
 
-func parseBGPAdvertisements(ads []bgpAdvertisement, cidrsPerAddresses map[string][]*net.IPNet, communities map[string]uint32) ([]*BGPAdvertisement, error) {
+func bgpAdvertisementsFromCR(ads []metallbv1beta1.BgpAdvertisement, cidrsPerAddresses map[string][]*net.IPNet, communities map[string]uint32) ([]*BGPAdvertisement, error) {
```

Comment:
Just an opinion: it would be nice to have a lint rule that limits
code lines to 120 chars. (this one is 173 chars long and runs off the screen)

Reply:
Yeah I don't know if / what go guidelines are, but makes sense I guess